runtime/util/hshelp.c

```diff
@@ -3187,10 +3187,20 @@ static jvmtiError
 verifyRecordAttributesAreSame(J9ROMClass *originalROMClass, J9ROMClass *replacementROMClass)
 {
 	jvmtiError rc = JVMTI_ERROR_NONE;
+#if JAVA_SPEC_VERSION >= 16
+	J9UTF8* originalSuperclassName = J9ROMCLASS_SUPERCLASSNAME(originalROMClass);
+	J9UTF8* replacementSuperclassName = J9ROMCLASS_SUPERCLASSNAME(replacementROMClass);
+#endif /* JAVA_SPEC_VERSION >= 16 */
 
 	/* Since retranformation is not allowed to change inheritance there's no need to consider 
 	 * one class being a record and one not. */
-	if (J9ROMCLASS_IS_RECORD(originalROMClass) && J9ROMCLASS_IS_RECORD(replacementROMClass)) {
+	if (J9ROMCLASS_IS_RECORD(originalROMClass)
+		&& J9ROMCLASS_IS_RECORD(replacementROMClass)
+#if JAVA_SPEC_VERSION >= 16
+		&& (0 == memcmp("java/lang/Record", J9UTF8_DATA(originalSuperclassName), J9UTF8_LENGTH(originalSuperclassName)))
```

Comment:
This seems unnecessary - you can't change the superclass during redefine, so both classes are subclasses of Record or neither is without doing this check. Same comment about reading undefined memory as above.

Reply:
Theoretically it is true in such case. I can remove the check in this file & `reflecthelp.c` and leave it unchanged to see what happens. For now there is no test suite related to this case.

Reply:
The check on superclass has been removed in this file.